### PR TITLE
Tag CUDA v0.3.0

### PR DESCRIPTION
**Pull request by maleadt**
This deprecates CUDA.jl (it had been documented for a while already), freeing up the name so that we can use it to host the actual Julia CUDA support in a couple of months.

cc @vchuravy

### CODE REVIEW

**Comment by ararslan** [comment]
Since you're tagging a new version, you'll also need a `require` file describing the dependencies and their versions, including the minimum supported version of Julia. That's what's causing the Travis failure.

**Comment by tkelman** [comment]
A deprecation would generally apply an upper julia version bound to all past tags of the package, making it uninstallable in the future. Are any other packages depending on this one? If you replace the content with something new, please do so via a merge commit instead of a force push so the old tags' history is still present.

**Comment by maleadt** [comment]
OK. Added upper bounds to all tags which point to functional versions (eg. [v0.2.2](https://github.com/JuliaGPU/CUDA.jl/commit/676ddc42460c760aa3b2c314b0da48d2535a91c4)), with only the last version -- ie. the one which calls `error(this package is deprecated)` upon loading -- installable on 0.4+.

Wrt. dependent packages: I know of none, and I didn't find any in METADATA.

**Comment by ararslan** [comment]
Yep, looks like there are no registered packages that depend on CUDA.

File: CUDA/versions/0.0.0/sha1 [hunk 1]
```diff
@@ -1 +1 @@
-3e81634d29c649fea5165fafaa9438072fc8f6be
+cccf07371a1e475badaef35d55a88768f898a286
```
Comment (tkelman) [inline CUDA/versions/0.0.0/sha1]: why are you modifying old sha's? you shouldn't do this

**Comment by maleadt** [comment]
I'm not following. Which of METADATA.jl/CUDA/versions/*/{requires,sha1} and/or CUDA.jl's tags do you want changed?

**Comment by tkelman** [comment]
Only the requires files in metadata should be modified by this PR. The tags and sha1's should have stayed as they were.

**Comment by maleadt** [comment]
I assumed they had to match up. Reverted.

**Comment by maleadt** [comment]
Thanks.

**Pull request merged**